osu.Game/Skinning/LegacySkin.cs

```diff
@@ -344,45 +343,29 @@ public override Drawable GetDrawableComponent(ISkinComponent component)
                                 }
                             })
                             {
-                                Children = new[]
-                                {
-                                    // TODO: these should fallback to the osu!classic skin.
-                                    GetDrawableComponent(new HUDSkinComponent(HUDSkinComponents.ComboCounter)) ?? new DefaultComboCounter(),
-                                    GetDrawableComponent(new HUDSkinComponent(HUDSkinComponents.ScoreCounter)) ?? new DefaultScoreCounter(),
-                                    GetDrawableComponent(new HUDSkinComponent(HUDSkinComponents.AccuracyCounter)) ?? new DefaultAccuracyCounter(),
-                                    GetDrawableComponent(new HUDSkinComponent(HUDSkinComponents.HealthDisplay)) ?? new DefaultHealthDisplay(),
-                                    GetDrawableComponent(new HUDSkinComponent(HUDSkinComponents.SongProgress)) ?? new SongProgress(),
-                                }
+                                Children = this.HasFont(LegacyFont.Score)
+                                    ? new Drawable[]
+                                    {
+                                        new LegacyComboCounter(),
+                                        new LegacyScoreCounter(),
+                                        new LegacyAccuracyCounter(),
+                                        new LegacyHealthDisplay(),
+                                    }
+                                    : new Drawable[]
+                                    {
+                                        // TODO: these should fallback to using osu!classic skin textures, rather than doing this.
```

Comment:
How do you envision this being possible with this change?

Reply:
The HUD components themselves will fetch the sprite text textures, as they'll be added inside the target container which is under all the required skin providing containers.

Reply:
To express further on this since I've stumbled on it again, the following diagram demonstrates what I'm thinking about having as a first step with the current new skinning system around (as I'm working on the default skin legacy fallback support FWIW):

![](https://user-images.githubusercontent.com/22781491/118770856-7b62c400-b88a-11eb-9517-df0ba379696d.png)

It uses "Does skin have at least one resource skinned from what's required" as a way of considering the skin to have the requested component explicitly skinned and the skin should return it instead of falling back, which is a thing that would be useful for beatmap skins as the `WorkingBeatmap` would just have to always instantiate the beatmap skin on `GetSkin()` and leave it for the resources of the beatmap to define whether the beatmap skin should be used or ignored.

And before the above check gets executed, `Skin.GetDrawableComponent` will be called first and if the skin has a customized set of the given target, then it'll immediately be returned instead.